src/App.tsx

```diff
@@ -1,9 +1,23 @@
-import React from 'react';
+import React, { useState } from 'react';
 import './App.scss';
 import { MoviesList } from './components/MoviesList';
 import moviesFromServer from './api/movies.json';
 
 export const App: React.FC = () => {
+  const [query, setQuery] = useState('');
+
+  const handleChange = (event: React.ChangeEvent<HTMLInputElement>) => {
+    setQuery(event.target.value);
+  };
+
+  const preparedQuery = query.toLowerCase();
+
+  const visibleMovies = !query
+    ? [...moviesFromServer]
+    : moviesFromServer
```

Comment:
Great job! The only thing I recomend to change is the way how you make your array visible. 
```query &&  array.filter(...)``` will do the same thing as the ternary operator in this case

Reply:
agree, filter should only be executed when query is not empty